### ISSUE TITLE
double free or corruption (out) core dump on large sized js enabled site.

### ISSUE THREAD

**Issue by PresGas** [newsboat/newsboat#1300]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: newsboat r2.21 

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
save-path "/home/rmday/Documents/saved-articles/"
#urls-source "ocnews"
#ocnews-url "https://cloud.disroot.org/"
#ocnews-login "presgas"
#ocnews-flag-star "s"
max-items 250
article-sort-order date-desc
auto-reload yes
reload-time 120
reload-threads 8


# Dark solarized color scheme for newsbeuter, based on
# Ethan Schoonover's Solarized.
#
# In order to use this color scheme, you must first configure
# your terminal emulator to use the Solarized palette.
# See <http://ethanschoonover.com/solarized/> for more information.

color background   default   default
color listnormal   default   default
color listnormal_unread   default   default
color listfocus    black     yellow
color listfocus_unread    black     yellow
color info         default   black
color article      default   default

# highlights
highlight article "^(Title):.*$" blue default
highlight article "https?://[^ ]+" red default
highlight article "\\[image\\ [0-9]+\\]" green default

ignore-article "*" "author =~ \"Greyhawk Grognard\""
ignore-article "*" "author =~ \"Timothy S. Brannan\""
ignore-article "*" "author =~ \"James V. West\""
#ignore-article "*" "link =~ MonstersAndManuals"

```

Steps to reproduce the issue:

1. Select article from rss feed with what looks like a large payload (js)

> Nov 22 281.4K  Finders Keepers

2. newsboat crashes 

3. double free or corruption (out)

**Other info you think is relevant**: The rss feed that I did get this from was a planet aggregator (https://campaignwiki.org/rpg/) with rss feed (https://campaignwiki.org/rpg/atom.xml). There have been other large looking articles that have done this, but this is the first I really tried to get info on. The actual article causing the crash is titled "Finders Keepers" and is here:

https://slightadjustments.blogspot.com/2020/08/finders-keepers-in-space.html

Many of the articles are using js for generators and this may well be the issue, but letting "the experts" look this over.

**Comment by Minoru** [comment]
Reproduced with r2.21-213-gef253 and empty config. It appears that this line pops from an empty vector:

https://github.com/newsboat/newsboat/blob/ef2531d7cc17f74e427c436b95330415c69a8368/src/htmlrenderer.cpp#L640

I'm minimizing the feed, and once I got a minimal test, I'll see how this should be fixed.

Thank you for the report, @PresGas!

**Comment by PresGas** [comment]
Thanks so much for the quick replication, @Minoru !

I've been a user since the early newsbeuter days. The project is in good hands.

**Comment by PresGas** [comment]
An additional data point. I noticed the same article got "updated" in the planet I am following. So some readers (of which a planet is a sort of reader aggregate), may occasionally "update" an old article and re-view it as unread. I know newsboat has that feature.

It seems to be the same article with a different size.

**Comment by Minoru** [comment]
Ah, sorry I didn't write about it here. I already tracked this bug down: it's caused by the HTML inside the \<script> tag. Newsboat doesn't properly skip the body of the script, gets confused by the HTML, and corrupts its own memory. This is fixed in #1302, which you can build and try out if you have all the necessary libraries and tools (see README).

If no other issues will be found with that pull request, it'll get merged in a few days and will make it into the December release.
